swarm/api/routes/evolution.py

```
@@ -225,6 +227,18 @@ async def get_run_evolution_patches(run_id: str):
     """
+    try:
+        run_id = validate_path_component(run_id, "run_id")
+    except ValueError as e:
+        raise HTTPException(
+            status_code=400,
+            detail={
+                "error": "invalid_path",
+                "message": str(e),
+                "details": {"run_id": run_id},
+            },
+        )
```

Ruff (0.15.4)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)